src/ray/gcs/redis_context.h

```diff
@@ -170,6 +187,7 @@ class RedisContext {
   }
 
  private:
+  boost::asio::io_service &io_service_;
```

Comment:
Hmmm, we wanted to avoid references as class members, right? The lifetime is not clear for them and it is very dangerous.

Reply:
The io_service here is the event loop for GcsRedisClient, and the lifetime of it is at least the same as this context, so the reference here is safe. But is better to add a comment here.

Reply:
@micafan Please *always* avoid references as class members. It's not merely a safety issue. It's quite annoying and has no actual benefit. For example, I had to track this down right now:

	src\ray/gcs/redis_context.h(87,5): error: explicitly defaulted default constructor is implicitly deleted [-Werror,-Wdefaulted-function-deleted]
	    CallbackItem() = default;
	    ^
	src\ray/gcs/redis_context.h(106,30): note: default constructor of 'CallbackItem' is implicitly deleted because field 'io_service_' of reference type 'boost::asio::io_service &' (aka 'boost::asio::io_context &') would not be initialized
	    boost::asio::io_service &io_service_;
				     ^

Reply:
I agreed that we should use pointer here. But in this case we should make the default constructor as deleted and make this struct as class. The reason here I agree we use reference is that we already use io_service reference

Reply:
They should be passed as reference parameters if they shouldn't be null, but they should be stored as pointers regardless. I'll fix it.

Reply:
#6285

Reply:
You can fix those class that make io_service as reference member in ray core.

Reply:
We should pass pointer parameters if we might modify it and pass const reference parameters if we don't modify it.

Reply:
@micafan Yeah, I need to fix the others too. For some reason I haven't gotten compile errors for them yet, so they haven't been as urgent, but hopefully I'll have them all fixed.